pil-analyzer/tests/types.rs

```diff
@@ -761,3 +761,22 @@ fn prover_functions() {
     ";
     type_check(input, &[("a", "", "int"), ("b", "", "()[]")]);
 }
+
+#[test]
+fn typed_literals() {
+    let input = "
+        let a = -1_int;
+        let b = -1_2_fe;
+        let c = -0x7_8_int;
+        let d = [1, 0_int, 2];
```

Comment:
Can you add a test without `_` between the number and the type?

Reply:
oh! And we need a display test!